frontend/console/src/features/modules/module.utils.ts

```diff
@@ -96,6 +126,44 @@ export const declFromSchema = (moduleName: string, declName: string, schema: Pul
   return module.schema.decls.find((d) => d.value.value?.name === declName)
 }
 
+type WithExport = { export?: boolean }
+
+export const declSumTypeIsExported = (d: DeclSumType) => {
+  return (d as WithExport).export === true
+}
+
+export const declFromModules = (moduleName: string, declCase: string, declName: string, modules?: Module[]) => {
+  if (!modules) {
+    return undefined
+  }
+  const module = modules.find((m) => m.name === moduleName)
+  if (!module) {
+    return undefined
+  }
+  switch (declCase) {
```

Comment:
might be nice to reduce the repetition here; would something like this be suitable?

const declMap: Record<string, keyof typeof Module> = {
  config: 'configs',
  data: 'data',
  database: 'databases',
  ...
};

const key = declMap[declCase];
if (key) {
  return module[key].find((id) => d[key]?.name === declName)?.[key];
}